srml/contracts/src/exec.rs

```diff
@@ -61,9 +61,11 @@ pub trait Ext {
 	/// was deleted.
 	fn get_storage(&self, key: &StorageKey) -> Option<Vec<u8>>;
 
-	/// Sets the storage entry by the given key to the specified value.
+	/// Sets the storage entry by the given key to the specified value. If `value` is `None` then
+	/// the storage entry is deleted.
 	///
-	/// If `value` is `None` then the storage entry is deleted.
+	/// Pre-conditions: The caller must check before that the value size is allowed using
+	/// `is_storage_len_allowed`, otherwise this method will panic.
```

Comment:
I think a better way is to return `Result` and then at the call site proof that there can't be `Err`.